### PR TITLE
7/Database connection

### PR DESCRIPTION
**Pull request by AlaaShurrab**
Relates #7 
creating the database files and the connection file

### CODE REVIEW

File: server/database/config/build.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  price FLOAT,
+  payment FLOAT,
+  balance FLOAT,
+  appointment_date TIME DEFAULT current_time
+);
+
+CREATE TABLE users(
+  id SERIAL PRIMARY KEY,
+  email VARCHAR(100) NOT NULL UNIQUE,
+  password VARCHAR(250) NOT NULL,
```
Comment (hassanelnajjar) [inline server/database/config/build.sql:37]: remove the comma from the end

File: server/database/config/build.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
+BEGIN;
+
+DROP TABLE IF EXISTS patients,appointments,history,users, CASCADE
```
Comment (hassanelnajjar) [inline server/database/config/build.sql:3]: add semicolon after CASCADE

File: server/database/config/build.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+CREATE TABLE patients(
+  id SERIAL PRIMARY KEY,
+  firstname VARCHAR(100) NOT NULL,
+  lastname VARCHAR(100) NOT NULL,
+  email VARCHAR (100),
+  birthday DATE,
+  phone VARCHAR(30) NOT NULL UNIQUE,
```
Comment (hassanelnajjar) [inline server/database/config/build.sql:11]: remove the comma from the end

File: server/database/config/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const connection = require('./connection');
+const dbBuild = require('./build');
+
```
Comment (hassanelnajjar) [inline server/database/config/index.js:3]: Add This Function here so it's will build db when execute `npm run build:db`
```js
(async () => {
  try {
    await dbBuild();
    console.log('Build Successfully');
    connection.end();
  } catch (e) {
    console.log('Build Failed', e);
  }
})();
```

**Review by hassanelnajjar — CHANGES_REQUESTED** [review]

File: server/database/config/index.js (hunk 4 above)
Comment (AlaaShurrab) [reply server/database/config/index.js:3]: done

File: server/database/config/databaseConfig.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+CREATE DATABASE dentorodev;
+CREATE USER dentorouser WITH superuser password '123456';
+ALTER DATABASE dentorodev OWNER TO dentorouser;
+-- test db
+CREATE DATABASE dentorotest;
+CREATE USER dentorouser2 WITH superuser password '123456';
+ALTER DATABASE dentorotest OWNER TO dentorouser2;
```
[the remark this answers is not in the document]
Comment (AlaaShurrab) [reply server/database/config/databaseConfig.sql]: done

File: server/database/config/connection.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    dbUrl = TEST_DB_URL;
+    break;
+
+  default:
+    throw new Error('There is no Database url !!!');
+}
+
+const option = {
+  connectionString: dbUrl,
+  ssl: !!process.env.DATABASE_URL,
```
[the remark this answers is not in the document]
Comment (AlaaShurrab) [reply server/database/config/connection.js]: done

**Review by Alaalser — APPROVED** [review]

File: .eslintrc.json [hunk 7]
```diff
@@ -16,6 +16,7 @@
 	},
 	"plugins": ["prettier", "jest"],
 	"rules": {
+		"no-console": "off",
```
Comment (Mu7ammadAbed) [inline .eslintrc.json:19]: Remove this rule :)

File: server/database/config/build.sql [hunk 8]
```diff
@@ -8,27 +8,26 @@ CREATE TABLE patients(
   lastname VARCHAR(100) NOT NULL,
   email VARCHAR (100),
   birthday DATE,
-  phone VARCHAR(30) NOT NULL UNIQUE
+  phone VARCHAR(30) NOT NULL UNIQUE,
+  diseses TEXT
```
Comment (Mu7ammadAbed) [inline server/database/config/build.sql:12]:
```suggestion
  diseases TEXT
```

File: server/database/config/build.sql [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+);
+
+CREATE TABLE appointments(
+  id SERIAL PRIMARY KEY,
+  patient_id INTEGER REFERENCES patients(id) ON UPDATE CASCADE,
+  appointment_date DATE NOT NULL,
+  appointment_time TIME NOT NULL,
+  is_done BOOLEAN DEFAULT false,
+  complaines TEXT
```
Comment (Mu7ammadAbed) [inline server/database/config/build.sql:21]:
```suggestion
  Complaints TEXT
```

File: server/database/config/build.sql [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+);
+
+CREATE TABLE history(
+  id SERIAL PRIMARY KEY,
+  appointment_id INTEGER REFERENCES appointments(id) ON UPDATE CASCADE,
+  descrption TEXT NOT NULL,
```
Comment (Mu7ammadAbed) [inline server/database/config/build.sql:27]:
```suggestion
  description TEXT NOT NULL,
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/database/config/fakeData.sql [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+(4,false,'2021-12-02','22:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+(2,false,'2021-12-02','22:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+(12,true,'2021-1-02','14:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+(12,false,'2021-12-02','12:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+(12,false,'2021-12-02','21:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+(2,false,'2020-12-02','20:30','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad');
+
+insert into history(appointment_id,price,payment,balance,descrption)
```
Comment (Mu7ammadAbed) [inline server/database/config/fakeData.sql:35]:
```suggestion
insert into history(appointment_id,price,payment,balance,description) 
```

File: server/database/config/fakeData.sql [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+('Adriel','Kozey','German_Kirlin27@gmail.com','1-574-799-5311 x41309','1996-12-02','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+('Isai','Block','Keshawn.Krajcik@gmail.com','(769) 704-9882 x942','1996-12-02','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+('Melvin','Haley','Abby_Hoppe53@yahoo.com','929-815-1747 x66522','1996-12-02','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+('Cristian','Wintheiser','Cecilia41@yahoo.com','(166) 430-8386','1996-12-02','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad'),
+('Milton','Konopelski','Josefina79@hotmail.com','1-700-991-9688','1996-12-02','Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad');
+
+insert into appointments(patient_id,is_done,appointment_date,appointment_time,complaines)
```
Comment (Mu7ammadAbed) [inline server/database/config/fakeData.sql:24]:
```suggestion
insert into appointments(patient_id,is_done,appointment_date,appointment_time,Complaints) 
```

File: server/database/config/fakeData.sql [hunk 13]
```diff
@@ -0,0 +1,42 @@
+insert into patients(firstname,lastname,email,phone,birthday,diseses)
```
Comment (Mu7ammadAbed) [inline server/database/config/fakeData.sql:1]:
```suggestion
insert into patients(firstname,lastname,email,phone,birthday,diseases) 
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 7 above)
Comment (AlaaShurrab) [reply .eslintrc.json]: done

File: .eslintrc.json [hunk 14]
```diff
@@ -16,7 +16,7 @@
 	},
 	"plugins": ["prettier", "jest"],
 	"rules": {
-		"no-console": "off",
+		// "no-console": "off",
```
Comment (Mu7ammadAbed) [inline .eslintrc.json:19]: Remove this line not commenting it

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 14 above)
Comment (AlaaShurrab) [reply .eslintrc.json]: done

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Pull request merged**